### ISSUE TITLE
RPC: UDFs not serialized

### ISSUE THREAD

**Issue by scsmithr** [GlareDB/glaredb#1697]
## Description

```
  2023-09-01T17:32:35.206140Z ERROR testing::slt::cli: Error while running test `sqllogictests/psql`, error: test fail: statement failed: Execution error: failed to execute physical plan on remote service: Remote session error: error while executing physical plan: status: Unknown, message: "Error during planning: There is no UDF named \"pg_table_is_visible\" in the registry", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Fri, 01 Sep 2023 17:32:35 GMT", "content-length": "0"} }
[SQL] SELECT n.nspname AS "Schema",
    c.relname AS "Name",
    CASE c.relkind WHEN 'r' THEN 'table'
    WHEN 'v' THEN 'view'
    WHEN 'm' THEN 'materialized view'
    WHEN 'i' THEN 'index'
    WHEN 'S' THEN 'sequence'
    WHEN 's' THEN 'special'
    WHEN 't' THEN 'TOAST table'
    WHEN 'f' THEN 'foreign table'
    WHEN 'p' THEN 'partitioned table'
    WHEN 'I' THEN 'partitioned index' END AS "Type",
    pg_catalog.pg_get_userbyid(c.relowner) AS "Owner"
FROM pg_catalog.pg_class AS c
    LEFT JOIN pg_catalog.pg_namespace AS n ON n.oid = c.relnamespace
    LEFT JOIN pg_catalog.pg_am AS am ON am.oid = c.relam
WHERE c.relkind IN ('r', 'p', 'v', 'm', 'S', 'f', '')
    AND n.nspname <> 'pg_catalog'
    AND n.nspname !~ '^pg_toast'
    AND n.nspname <> 'information_schema'
    AND pg_catalog.pg_table_is_visible(c.oid)
ORDER BY 1, 2;
at /Users/sean/Code/github.com/glaredb/glaredb/testdata/sqllogictests/psql.slt:4

```

**Comment by universalmind303** [comment]
@scsmithr I've been looking into this one & It seems like there is significantly more work than I anticipated. Initially I thought we could just throw them into datafusion's `FunctionRegistry` and call it a day.  While this works for most of them, several rely on `session_vars`. 

This is pretty straightforward for functions that don't rely on the context such as `pg_get_userbyid` and `pg_table_is_visible`, but ones that are dependent on it such as `current_database` and `current_schema` are a bit trickier. 



I was thinking of implementing a physicalExpr for those functions that takes in the ctx vars as function arguments, then implementing `create_physical_expr` in our planner. The `PhysicalPlanner` is the only point during execution/planning that we have access to the sessionvars, so i think we'd have to do the mapping there. 


Was wondering if you had any thoughts on other potential solutions?

**Comment by scsmithr** [comment]
I don't really have any thoughts on this. But I figured there might be some tricky bits due to that dependency.

Implementing a physical expr seems like it'd make sense. I say we go with a quick proof-of-concept for one of the functions that we're having trouble with, and see how that plays out.

**Comment by universalmind303** [comment]
closed via #1755
